py-rse/style.Rmd

```diff
@@ -4,220 +4,272 @@
+        for col in range(original.numCol):
+            result[row, col] = original[col, row]
+    return result
+```
 
-Ask for clarification.
-:   If you don't understand something,
-    or don't understand why the author did it,
-    ask.
-    (And when the author explains it,
-    think about suggesting that the explanation ought to be documented somewhere.)
+The use of 4 spaces is a compromise between 2
+(which we find perfectly readable, but some people find too crowded)
+and 8
+(which most people agree uses up too much horizontal space).
+As for the use of spaces rather than tabs,
+the original reason was that the most common interpretation of tabs
+by the editors of the 1980s was 8 spaces,
+which again was more than most people felt necessary.
+Today,
+almost all editors will auto-indent or auto-complete when the tab key is pressed,
+but the legacy of those times lives on.
+
+**Do not put spaces inside parentheses.**
+
+Don't write <code>(&nbsp;1+2&nbsp;)</code>.
+This applies to function calls as well:
+write `max(a, b)` rather than <code>max(&nbsp;a,&nbsp;b&nbsp;)</code>.
 
-Offer alternatives.
-:   Telling authors that something is wrong is helpful;
-    telling them what they might do instead is more so.
+**Always use spaces around comparisons like `>` and `<=`.**
 
-Don't be sarcastic or disparaging.
-:   "Did you maybe think about *testing* this garbage?"
-    is a Code of Conduct violation in any well-run project.
+Python automatically interprets `a+b<c+d` as `(a+b)<(c+d)`,
+but that's a lot of punctuation crowded together;
+using spaces around comparison operators makes it easier to see
+what's being compared to what.
+However,
+you should use your own judgment for spacing around arithmetic operators like `+` and `/`.
+For example,
+`a+b+c` is perfectly readable,
+but
 
-Don't present opinions as facts.
-:   "Nobody uses X any more" might be true.
-    If it is,
-    the person making the claim ought to be able to point at download statistics
-    or a Google Trends search;
-    if they can't,
-    they should say,
-    "I don't think anybody uses X any more" and explain why they think that.
+```python
+substrate[i, j] + overlay[i, j]
+```
 
-Don't feign surprise or pass judgment.
-:   "Gosh, didn't you know [some obscure fact]?" isn't helpful;
-    neither is, "Geez, why don't you [some clever trick] here?"
+is probably easier for the eye to follow
+than the spaceless:
 
-Don't overwhelm people with details.
-:   If someone has used the letter `x` as a variable name in several places,
-    and they shouldn't have,
-    comment on the first two or three and simply put a check beside the others—the reader
-    won't need the comment repeated.
+```python
+substrate[i, j]+overlay[i, j]
+```
 
-Don't ask people to do extra work.
-:   Nobody enjoys fixing bugs and style violations.
-    Asking them to add a few features while they're at it is rude.
+**Put two blank links between each function definition.**
```

Comment:
I took it out because it added a lot of space on the page without the readability difference being clear - I think it only pays off once you have many functions.